### PR TITLE
fix(TDP-2663): Hook default locale set higher in startup sequence.

### PR DESCRIPTION
**Pull request by fhuaulme**
**Link to the JIRA issue**
https://jira.talendforge.org/browse/TDP-2663

**Please check if the PR fulfills these requirements**
- [x] The commit(s) message(s) follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md#commit-message-format)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [ ] The code coverage on new code is > 75 % for backend and > 95% for frontend
- [x] The new code does not introduce new technical issues (sonar / eslint)
- [ ] Functional tests have been performed
- [ ] Docker configuration files for config-std or config-cloud profiles are impacted

**Please check the browsers you've tested on**
- [ ] Chrome, Firefox, Safari, Edge, IE11
- [ ] No, that's bad, this PR should not be merged !
- [x] No, and no need to (backend changes only)

* Hook default locale setup during bean scan so it gets set *before* bean initialization.

### CODE REVIEW

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1218/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1218/)

File: dataprep-backend-service/src/main/java/org/talend/dataprep/configuration/Localization.java [hunk 1]
```diff
@@ -17,43 +17,58 @@
...
+        private static final Logger LOGGER = getLogger(Localization.class);
+
+        @Override
+        public void registerBeanDefinitions(AnnotationMetadata importingClassMetadata,
+                                            BeanDefinitionRegistry registry) {
+            if (registry instanceof BeanFactory) {
+                final Environment environment = ((BeanFactory) registry).getBean(Environment.class);
+                Locale locale;
+                final String defaultLocale = environment.getProperty("dataprep.default.locale", "en-US");
+                if (StringUtils.isEmpty(defaultLocale)) {
```
Comment (elogeart) [inline dataprep-backend-service/src/main/java/org/talend/dataprep/configuration/Localization.java]: The condition `StringUtils.isEmpty(defaultLocale)` can never be true.

Comment (fhuaulme) [reply dataprep-backend-service/src/main/java/org/talend/dataprep/configuration/Localization.java]: Indeed, fixing this.

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1220/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1220/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [failure](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1221/) [:x:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1221/)

**Comment by build-talend-tdp** [comment]
Pull request test result: [success](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1222/) [:star:](https://ci-tdp.datapwn.com/job/data-prep_pull-request_master/1222/)

**Pull request merged**